velocity/src/main/java/me/leoko/advancedban/velocity/VelocityMethods.java

```diff
@@ -0,0 +1,377 @@
+      }
+
+      return json.get(keys[keys.length - 1]).toString().replaceAll("\"", "");
+
+    } catch (Exception exc) {
+      return null;
+    }
+  }
+
+  @Override
+  public String getVersion() {
+    return "@version";
+  }
+
+  @Override
+  public String[] getKeys(Object file, String path) {
+    return ((ConfigurationNode)file).getNode(path.split("\\.")).getChildrenMap().keySet().stream().map(Object::toString).toArray(String[]::new);
+  }
+
+  @Override
+  public void setupMetrics() {
+    // no bstats for velocity yet
+  }
+
+
+  @Override @Deprecated
+  public boolean isBungee() {
+    return false;
+  }
+
+  @Override
+  public boolean isProxy() {
+    return true;
+  }
+
+  @Override
+  public ServerType getServerType() {
+    return ServerType.VELOCITY;
+  }
+
+  @Override
+  public String clearFormatting(String text) {
+    return text.replaceAll("(&[^\\s])+", "");
+  }
+
+  @Override
+  public Object getPlugin() {
+    return VelocityMain.get();
+  }
+
+  @Override
+  public File getDataFolder() {
+    return dataDirectory.toFile();
+  }
+
+  @Override
+  public void setCommandExecutor(String cmd, TabCompleter tabCompleter) {
+    CommandMeta meta = server.getCommandManager().metaBuilder(cmd).build();
+    server.getCommandManager().register(meta, new CommandRecieverVelocity(server, cmd));
+  }
+
+  @Override
+  public void sendMessage(Object player, String msg) {
+      ((CommandSource) player).sendMessage(LegacyComponentSerializer.legacyAmpersand().deserialize(msg));;
+  }
+
+  @Override
+  public String getName(Object player) {
+    if (player instanceof Player) {
+      return ((Player) player).getUsername();
+    } else {
+      return "CONSOLE";
+    }
+  }
+
+  @Override
+  public String getName(String uuid) {
+      return server.getPlayer(UUID.fromString(uuid)).get().getUsername();
+  }
+
+  @Override
+  public String getIP(Object player) {
+    return ((Player)player).getRemoteAddress().getAddress().getHostAddress();
+  }
+
+  @Override
+  public String getInternUUID(Object player) {
+    return player instanceof  Player ? ((Player) player).getUniqueId().toString().replace("-", "") : "none";
+  }
+
+  @Override
+  public String getInternUUID(String player) {
+    Optional<Player> optionalPlayer = server.getPlayer(player);
+    return optionalPlayer.map(value -> value.getUniqueId().toString().replace("-", "")).orElse(null);
+  }
+
+  @Override
+  public boolean hasPerms(Object player, String perms) {
+      return ((CommandSource) player).hasPermission(perms);
+  }
+
+  @Override
+  public Permissionable getOfflinePermissionPlayer(String name) {
+    if(luckPermsSupport) return new LuckPermsOfflineUser(name);
+
+    return permission -> false;
+  }
+
+  @Override
+  public boolean isOnline(String name) {
+    return server.getPlayer(name).isPresent();
+  }
+
+  @Override
+  public Player getPlayer(String name) {
+    if(server.getPlayer(name).isPresent()) {
+      return server.getPlayer(name).get();
```

Comment:
```suggestion
      return server.getPlayer(name).orElse(null;
```
Would make the code cleaner and also avoid a double map lookup